src/gsCore/gsPolyField.h

```diff
@@ -0,0 +1,193 @@
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): A. Shamanskiy
+*/
+
+#pragma once
+
+#include <gsCore/gsFunctionSet.h>
+#include <gsCore/gsMultiPatch.h>
+#include <gsCore/gsGeometry.h>
+
+namespace gismo
+{
+
+/**
+ * \brief An object containing several scalar or vector fields defined on the same geometry.
+ *
+ * The class extends the idea behind gsField to a case of several fields. The primary purpose
+ * of the class is to serve as a convinient container which can be used to plot several
+ * physical fields defined on the same geometry to one Paraview file which can be useful
+ * for problems like incompressible Navier-Stokes (velocity + pressure), elasticity (deformation + stresses),
+ * thermal expansion (temperature + deformation), etc. Can be passed as an input argument to gsWriteParaview.
+ *
+ * \ingroup Core
+ */
+template <class T>
+class gsPolyField
+{
+
+public:
+
+    gsPolyField(const gsGeometry<T> & geo)
+        : m_patches(memory::make_shared_not_owned(&geo))
+    {}
+
+    gsPolyField(const gsMultiPatch<T> & patches)
+        : m_patches(memory::make_shared_not_owned(&patches))
+    {}
+
+    //=========================================================================//
+
+    void addField(const std::string & name, const gsMultiPatch<T> & f)
+    {
+        GISMO_ASSERT(std::find(m_names.begin(),m_names.end(),name) == m_names.end(),
+                     "Field with the name *" + name + "* already exists.\n");
+        GISMO_ASSERT(nPatches() == f.nPatches(),
+                     "The geometry has " + util::to_string(nPatches()) + " patch(es) and field *"
+                     + name + "* has " + util::to_string(f.nPatches()) + " patch(es).\n");
+        m_fields.push_back(memory::make_shared_not_owned(&f));
+        m_names.push_back(name);
+        m_isparams.push_back(true);
+    }
+
+    void addField(const std::string & name, const gsFunctionSet<T> & f, const bool isparam = false)
+    {
+        GISMO_ASSERT(std::find(m_names.begin(),m_names.end(),name) == m_names.end(),
+                     "Field with the name *" + name + "* already exists.\n");
+        GISMO_ASSERT(nPatches() == f.nPieces(),
+                     "The geometry has " + util::to_string(nPatches()) + " patch(es) and field *"
+                     + name + "* has " + util::to_string(f.nPieces()) + " patch(es).\n");
+        m_fields.push_back(memory::make_shared_not_owned(&f));
+        m_names.push_back(name);
+        m_isparams.push_back(isparam);
+    }
+
+    /// \brief Number of patches in the geometry
+    index_t nPatches() const { return m_patches->nPieces(); }
+
+    /// \brief Number of fields in the container
+    index_t nFields() const { return m_fields.size(); }
```

Comment:
This should be changed to `size_t`. See https://github.com/gismo/internal/issues/2